azurerm/data_source_kubernetes_cluster.go

```diff
@@ -564,6 +587,22 @@ func flattenKubernetesClusterDataSourceAgentPoolProfiles(input *[]containerservi
 			agentPoolProfile["count"] = int(*profile.Count)
 		}
 
+		if profile.MinCount != nil {
+			agentPoolProfile["min_count"] = int(*profile.MinCount)
+		}
+
+		if profile.MaxCount != nil {
+			agentPoolProfile["max_count"] = int(*profile.MaxCount)
+		}
+
+		if profile.EnableAutoScaling != nil {
+			agentPoolProfile["enable_auto_scaling"] = *profile.EnableAutoScaling
+		}
+
+		if profile.AvailabilityZones != nil {
+			agentPoolProfile["availability_zones"] = utils.FlattenStringSlice(profile.AvailabilityZones)
+		}
```

Comment:
could we remove this `if` statement here? the Flatten function handles this being nil and returning an empty list

Reply:
This is the other side of not sending empty lists to the API, I don't want to store the empty entry en the map so later I don't actually send it when invoking the api

Reply:
This shouldn't matter as its a datasource (and in the expand function you check for len >0 anyways)

Reply:
Agreed, as stated in the other comment all fields were following this pattern. Removed already